opslevel/datasource_opslevel_teams_all.go

```diff
@@ -0,0 +1,102 @@
+package opslevel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ datasource.DataSourceWithConfigure = &TeamDataSourcesAll{}
+
+func NewTeamDataSourcesAll() datasource.DataSource {
+	return &TeamDataSourcesAll{}
+}
+
+type TeamDataSourcesAll struct {
+	CommonDataSourceClient
+}
+
+type teamDataSourcesAllModel struct {
+	Teams []teamDataSourceModel `tfsdk:"teams"`
+}
+
+func newTeamDataSourcesAllModel(teams []opslevel.Team) teamDataSourcesAllModel {
+	teamModels := make([]teamDataSourceModel, 0)
+	for _, team := range teams {
+		teamModel := newTeamDataSourceModel(team)
+		teamModels = append(teamModels, teamModel)
+	}
+	return teamDataSourcesAllModel{Teams: teamModels}
+}
+
+func (d *TeamDataSourcesAll) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_teams"
+}
+
+func (d *TeamDataSourcesAll) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "List of all Team data sources",
+
+		Attributes: map[string]schema.Attribute{
+			"teams": schema.ListNestedAttribute{
+				NestedObject: schema.NestedAttributeObject{
```

Comment:
This was close a few commits back. Each of the updated datasources (plural) fits a local convention like:
```go
			"teams": schema.ListNestedAttribute{
				NestedObject: schema.NestedAttributeObject{
					Attributes: teamSchemaAttrs,
				},
				Description: "List of Team data sources",
				Computed:    true,
			},
```

`teamSchemaAttrs` itself should contain the attributes that fit both `team` and `teams` datasources. The attributes that are unique to `team` and unique to `teams` can both be separately fed into a `func TeamAttributes(attrs map[string]schema.Attribute) map[string]schema.Attribute { ... }` function.
See [example of this for domains](https://github.com/OpsLevel/terraform-provider-opslevel/blob/main/opslevel/datasource_opslevel_domain.go#L27-L56)

Reply:
@davidbloss I added `teamSchemaAttrs` in a new commit, and am excluding `func TeamAttributes` because the fields for the plural and singular datasource are the same (updated tophatting too for demonstration.)

Reply:
@davidbloss nevermind - realized an error when it comes to `Optional`